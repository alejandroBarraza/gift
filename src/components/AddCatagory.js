import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCatagory = ({ setcategories }) => {
    const [inputValue, setinputValue] = useState('');

    //function update input text
    const handleText = ({ target: { value } }) => {
        setinputValue(value);
    };

    //function update input value in the list
    const handleSubmit = (e) => {
        e.preventDefault();
        setcategories((cat) => [inputValue,...cat]);
        setinputValue('');
    };

    return (
        <form onSubmit={handleSubmit}>
            <h2>Add a new category</h2>
            <input type='text' required value={inputValue} onChange={handleText} />
        </form>
    );
};

AddCatagory.propTypes = {
    setcategories: PropTypes.func.isRequired,
};
