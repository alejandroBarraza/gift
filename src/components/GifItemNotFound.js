import React from 'react'

export const GifItemNotFound = ({category}) => {
    return (
        <div>
            <h2>{category} no encontrada, </h2>
            
        </div>
    )
}
