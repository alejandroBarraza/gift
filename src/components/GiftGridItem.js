import React from 'react';
import PropTypes from 'prop-types';

export const GiftGridItem = ({ url, title }) => {
    return (
        <div className='card glassmorphismo'>
            <p>{title}</p>
            <img src={url} alt={title} />
        </div>
    );
};

GiftGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
};
