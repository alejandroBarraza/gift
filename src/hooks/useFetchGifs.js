import { useState,useEffect } from "react"
import { getGift } from "../helpers/getGifts"

//custom hooks return state(data from api and boolen load)
export const useFetchGifs = (category) => {
    const [state, setstate] = useState({
        data:[],
        loading:true
    })

    useEffect(() => {
        getGift(category)
            .then((imgs)=>{
                setstate({
                    data:imgs,
                    loading:false
                })
            })
        
    }, [category])

    return state
   
}
