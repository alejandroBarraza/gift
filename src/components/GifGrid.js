
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';

import { GiftGridItem } from './GiftGridItem';



export const GifGrid = ({ category }) => {


    const {data:imgs} = useFetchGifs(category)
    

    return (
        <>
            <h2>busqueda ralizada : {category}</h2>
           
            <div className='card-container'>
                {
                    imgs.map( (img) => <GiftGridItem key={img.id} {...img} />)
                }   
            </div>
        </>
    );
};
