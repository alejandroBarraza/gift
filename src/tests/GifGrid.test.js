import '@wojtekmaj/enzyme-adapter-react-17';
import '../setupTests';
import { shallow } from 'enzyme';
import { GifGrid } from '../components/GifGrid';
import { useFetchGifs } from '../hooks/useFetchGifs';
import '@testing-library/jest-dom';
import { GiftGridItem } from '../components/GiftGridItem';
jest.mock('../hooks/useFetchGifs');

describe('testing <GifGrid /> component', () => {
    const category = 'naruto';
    let wrapper;
    beforeEach(() => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        wrapper = shallow(<GifGrid category={category} />);
    });
    afterEach(() => {
        wrapper.unmount();
    });
    test('should match <GifGrid category={cateogory}/>  with snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('should load item after use fetch in customHook UseFetchGifts', () => {
        const imgs = [
            {
                id: 13123,
                title: category,
                url: 'htttps://www.gift/naruto.png',
            },
        ];

        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false,
        });

        wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find(GiftGridItem)).toHaveLength(1);
    });
});
