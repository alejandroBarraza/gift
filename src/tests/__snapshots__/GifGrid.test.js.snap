// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <GifGrid /> component should load item after use fetch in customHook UseFetchGifts 1`] = `
<Fragment>
  <h2>
     
    naruto
  </h2>
  <div
    className="card-container"
  >
    <GiftGridItem
      id={13123}
      key="13123"
      title="naruto"
      url="htttps://www.gift/naruto.png"
    />
  </div>
</Fragment>
`;

exports[`testing <GifGrid /> component should match <GifGrid category={cateogory}/>  with snapshot 1`] = `
<Fragment>
  <h2>
     
    naruto
  </h2>
  <div
    className="card-container"
  />
</Fragment>
`;
